lib/getPostcssResult.mjs

```diff
@@ -65,22 +65,23 @@ export default async function getPostcssResult(stylelint, options = {}) {
 /**
  * @param {CustomSyntax} customSyntax
  * @param {string | undefined} basedir
- * @returns {Syntax}
+ * @returns {Promise<Syntax>}
  */
-function getCustomSyntax(customSyntax, basedir) {
+async function getCustomSyntax(customSyntax, basedir) {
 	if (typeof customSyntax === 'string') {
 		const customSyntaxLookup = basedir ? getModulePath(basedir, customSyntax) : customSyntax;
 
 		let resolved;
 
 		try {
-			resolved = require(customSyntaxLookup);
+			resolved = await import(customSyntaxLookup);
+			resolved = resolved.default ?? resolved;
 		} catch (error) {
 			if (
 				error &&
 				typeof error === 'object' &&
 				'code' in error &&
-				error.code === 'MODULE_NOT_FOUND' &&
+				(error.code === 'MODULE_NOT_FOUND' || error.code === 'ERR_MODULE_NOT_FOUND') &&
```

Comment:
[note] The path of `error.code === 'ERR_MODULE_NOT_FOUND'` is not covered in tests for some reason. I guess Babel under Jest may transform the `import()` code, but I'm not sure.